subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -1448,24 +1478,73 @@ static int nrf_cloud_parse_cell_pos_json(const cJSON *const cell_pos_obj,
 	location_out->lon = lon->valuedouble;
 	location_out->unc = (uint32_t)unc->valueint;
 
-	if (json_item_string_exists(cell_pos_obj, NRF_CLOUD_JSON_FULFILL_KEY,
-				    NRF_CLOUD_CELL_POS_TYPE_VAL_MCELL)) {
-		location_out->type = CELL_POS_TYPE_MULTI;
+	location_out->type = CELL_POS_TYPE__INVALID;
+
+	if (!get_string_from_obj(cell_pos_obj, NRF_CLOUD_JSON_FULFILL_KEY, &type)) {
+		if (!strcmp(type, NRF_CLOUD_CELL_POS_TYPE_VAL_MCELL)) {
+			location_out->type = CELL_POS_TYPE_MULTI;
+		} else if (!strcmp(type, NRF_CLOUD_CELL_POS_TYPE_VAL_SCELL)) {
+			location_out->type = CELL_POS_TYPE_SINGLE;
+		} else {
+			LOG_WRN("Unhandled cellular positioning type: %s", log_strdup(type));
+		}
 	} else {
-		location_out->type = CELL_POS_TYPE_SINGLE;
+		LOG_WRN("Cellular positioning type not found in message");
 	}
 
 	return 0;
 }
 
+int nrf_cloud_handle_error_message(const char *const buf,
+				   const char *const app_id,
+				   const char *const msg_type,
+				   enum nrf_cloud_error * const err)
+{
+	if (!buf || !err) {
+		return -EINVAL;
+	}
+
+	int ret;
+	cJSON *root_obj;
+
+	*err = NRF_CLOUD_ERROR_NONE;
+
+	root_obj = cJSON_Parse(buf);
+	if (!root_obj) {
+		LOG_DBG("No JSON found");
+		return -ENOMSG;
+	}
+
+	ret = get_error_code_value(root_obj, err);
+	if (ret) {
+		goto clean_up;
+	}
+
+	/* If provided, check for matching app id and msg type */
+	if (msg_type &&
```

Comment:
Would it make more sense to check for any provided msg type and app_id before getting the error code above?

Reply:
its a feature!
```
* @retval -ENOENT Error code found, but did not match specified app_id and msg_type.
```

Reply:
OK, I believe you.

Reply:
I didn't look closely enough to see that `err` is a pointer parameter that `get_error_code_value()` updates, and that `ret` is non-zero only if it has trouble actually getting the error code.